### PR TITLE
feat(constitution): complete wave 4 assurance expansion

### PR DESCRIPTION
**Pull request by jamesryancooper**
## What
Complete Wave 4 of the unified execution constitution and fix the follow-up CI regressions so the canonical Wave 4 state is fully represented in the remote branch.

## Why
Wave 4 has to be complete in both implementation and remote validation state so consequential runs and system-level claims remain replayable, interpretable, and evidence-gated.

No-Issue: governed harness constitutional maintenance

## How
Promoted the assurance and disclosure contract families, added the proof-plane, lab, and observability surfaces, extended canonical run evidence with RunCard and HarnessCard disclosure, then closed the remaining Wave 4 gaps with maintainability proof, evaluator routing, benchmark disclosure assets, and a generic Wave 4 backfill path. Fixed the failing PR checks by repairing the autonomy-policy workflow jq expression, tracking the canonical run-control/evidence files that CI validates, and publishing this superseding PR with metadata that conforms to repository policy.

## Profile Selection Receipt
- Semantic version source(s): `/.octon/octon.yml`
- Release state (`pre-1.0` or `stable`): `pre-1.0`
- `change_profile` (`atomic` or `transitional`): `transitional`
- Hard-gate facts (downtime, coordination, migration/backfill, rollback, blast radius, compliance): broad constitutional, assurance, lab, observability, retained-evidence, and generated-read-model coordination; canonical run evidence backfill required; rollback must not leave partial proof/disclosure state; support claims remain subordinate to support targets
- Tie-break status: `transitional` because Wave 4 required staged proof/disclosure promotion and canonical run evidence backfill
- `transitional_exception_note` (required when `pre-1.0` + `transitional`): recorded in the Wave 4 migration receipts under `/.octon/instance/cognition/context/shared/migrations/2026-03-27-assurance-lab-disclosure-expansion-cutover/plan.md` and `/.octon/instance/cognition/context/shared/migrations/2026-03-27-wave4-gap-remediation-cutover/plan.md`

## Implementation Plan
- Workstreams and scope:
  - promote constitutional assurance and disclosure families
  - add functional, behavioral, maintainability, recovery, and evaluator proof-plane surfaces
  - add top-level lab and observability domains plus reusable evaluator/HarnessCard tooling
  - extend canonical run evidence with replay, proof, measurement, intervention, and disclosure families
  - seed and validate canonical support and benchmark disclosure artifacts
  - fix PR/autonomy workflow and tracked run artifacts so CI validates the same state as local runs
- Public interfaces/types/contracts affected:
  - `framework/constitution/contracts/{assurance,disclosure}/**`
  - `framework/lab/**`
  - `framework/observability/**`
  - `framework/orchestration/runtime/_ops/scripts/write-run.sh`
  - `state/control/execution/runs/**`
  - `state/evidence/{runs,lab}/**`
- Test scenarios:
  - Wave 4 validator pass
  - run validator pass for the seeded canonical runs
  - full harness alignment pass
  - PR/autonomy policy rerun against fixed metadata and workflow quoting
- Assumptions/defaults:
  - support claims remain bounded to declared support-target tuples
  - historical evidence-only runs remain out of scope for automatic authority reconstruction

## Impact Map (code, tests, docs, contracts)
- Code:
  - `.github/workflows/pr-autonomy-policy.yml`
  - `.octon/framework/orchestration/runtime/_ops/scripts/write-run.sh`
  - `.octon/framework/lab/runtime/_ops/scripts/write-harness-card.sh`
  - `.octon/framework/assurance/evaluators/runtime/_ops/scripts/write-evaluator-review.sh`
- Tests/validators:
  - `.octon/framework/assurance/runtime/_ops/scripts/validate-assurance-disclosure-expansion.sh`
  - `.octon/framework/orchestration/runtime/runs/_ops/scripts/validate-runs.sh`
  - `.octon/framework/assurance/runtime/_ops/scripts/{validate-harness-structure.sh,validate-architecture-conformance.sh,alignment-check.sh}`
- Docs:
  - `.octon/README.md`
  - `.octon/instance/bootstrap/START.md`
  - `.octon/framework/assurance/README.md`
  - `.octon/framework/cognition/_meta/architecture/specification.md`
- Contracts:
  - `.octon/framework/constitution/contracts/{assurance,disclosure}/**`
  - `.octon/framework/constitution/contracts/objective/run-contract-v1.schema.json`
  - `.octon/framework/cognition/_meta/architecture/contract-registry.yml`

## Compliance Receipt
- [x] Selected exactly one execution profile before planning/implementation.
- [x] Applied release-maturity gate from semantic version.
- [x] Enforced pre-1.0 default (`atomic`) unless hard gates required `transitional`.
- [x] Included `transitional_exception_note` when required.
- [x] Included tie-break escalation when applicable.
- [x] Updated impacted contracts/docs/tests.
- [x] Honored charter change-control constraint (no direct principles charter edits without override evidence).

## Exceptions/Escalations
none

## Tradeoffs
Wave 4 is complete for the promoted canonical run-root model, but historical evidence-only run directories still require human reconstruction if they ever need full run-contract-era disclosure. The PR also includes validation-generated publication and generated read-model updates because the passing harness gate refreshes those surfaces.

## Testing
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-assurance-disclosure-expansion.sh`
- `bash .octon/framework/orchestration/runtime/runs/_ops/scripts/validate-runs.sh`
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-harness-structure.sh`
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-architecture-conformance.sh`
- `bash .octon/framework/assurance/runtime/_ops/scripts/alignment-check.sh --profile harness`
- `bash .octon/framework/assurance/runtime/_ops/scripts/validate-continuity-memory.sh`
- `git diff --check`

## Rollout
`n/a` for product rollout. Governance rollout is already reflected in the committed migration receipts, seeded canonical runs, and refreshed generated/publication outputs.

## Convivial Purpose Check
- [x] Feature expands genuine user capability (not synthetic engagement).
- [x] Attention and interruption behavior are justified and user-controllable.
- [x] No manipulative patterns or dark-pattern mechanics are introduced.
- [x] Data collection/extraction risk is minimal and explicitly justified.

## Risk Rubric
- Risk class: [ ] Trivial [x] Low [ ] Medium [ ] High
- Rollback plan: revert the Wave 4 commits and restore the prior constitutional/run-evidence state
- Rollback handle: `git revert` of the Wave 4 commits on the PR branch
- Flags changed (name, owner, expiry, rollout): none
- Autonomy eligibility: [ ] autonomy:auto-merge [x] autonomy:no-automerge

## Contracts and Threat Model
- OpenAPI/JSON-Schema changes:
  - added/updated constitutional assurance and disclosure schemas plus run-contract shape
- Threat model update/link:
  - disclosure remains subordinate to durable authority; support claims remain bounded to support-target declarations; evaluator review is explicit for higher-scrutiny claims

## Observability and Performance
- Traces/logs/metrics for changed flows:
  - retained measurements, interventions, replay manifests, RunCards, and HarnessCards under canonical `state/evidence/**` roots
- Representative traces for high risk changes:
  - canonical run evidence bundles for `run-wave3-runtime-bridge-20260327` and `run-wave4-benchmark-evaluator-20260327`
- Performance or bundle impact:
  - increased retained evidence volume for the seeded canonical runs; no material runtime binary or application performance change

## License and Provenance
- New dependencies and licenses:
  - none
- Generated code/templates provenance notes:
  - generated effective and cognition outputs plus publication receipts were refreshed by the passing harness validation flow

## Checklist
- [x] Requirements met; edge cases handled
- [x] Security reviewed (authz, input validation, secrets)
- [x] Tests added or updated
- [x] Observability updated (logs, metrics, traces) if needed
- [x] No speculative abstractions or unnecessary complexity
- [x] Conventions followed; no drift introduced
- [x] Non-obvious decisions documented (comments, ADR)
- [x] All review conversations resolved

## Screenshots / Notes
No UI screenshots. This PR updates constitutional/runtime/evidence surfaces and includes validation-generated artifacts that reflect the passing harness state.

### CODE REVIEW

File: .octon/framework/orchestration/runtime/_ops/scripts/write-run.sh [hunk 1]
```diff
@@ -813,6 +1262,9 @@ case "$cmd" in
         --lease-seconds) lease_seconds="$2"; shift 2 ;;
         --recovery-status) recovery_status="$2"; shift 2 ;;
         --recovery-reason) recovery_reason="$2"; shift 2 ;;
+        --risk-class) risk_class="$2"; shift 2 ;;
+        --reversibility-class) reversibility_class="$2"; shift 2 ;;
+        --support-tier) support_tier="$2"; shift 2 ;;
```
[the remark this answers is not in the document]
Comment (jamesryancooper) [reply .octon/framework/orchestration/runtime/_ops/scripts/write-run.sh:1308]: Added explicit fail-closed validation for `--risk-class` and `--support-tier` in `write-run.sh create`. Risk now accepts only `low|medium|high|critical`, and support tier must match a declared workload tier label or id from `/.octon/instance/governance/support-targets.yml`, so typos no longer fall through to `not_required` evaluator routing.

File: .octon/framework/orchestration/runtime/_ops/scripts/write-run.sh [hunk 2]
```diff
@@ -499,6 +654,281 @@ ensure_run_lifecycle_roots() {
...
+    "Recovery posture remains reconstructable from canonical rollback, checkpoint, and replay surfaces." \
+    "$(jq -cn --arg rollback "$(rollback_posture_relpath "$run_id")" --arg checkpoint "$(control_checkpoint_relpath "$run_id")" --arg replay "$(replay_manifest_relpath "$run_id")" '[$rollback,$checkpoint,$replay]')" \
+    "$recorded_at"
+  write_evaluator_review_file "$run_id" "$recorded_at"
+  write_measurement_summary_file "$run_id" "$recorded_at"
+  write_intervention_log_file "$run_id" "$recorded_at"
+  if [[ -n "$decision_id" ]]; then
+    write_run_card_file "$run_id" "$status" "$summary" "$recorded_at" "$decision_id"
+  fi
+  write_retained_evidence_file "$run_id" "$recorded_at"
```
[the remark this answers is not in the document]
Comment (jamesryancooper) [reply .octon/framework/orchestration/runtime/_ops/scripts/write-run.sh:972]: Updated the retained-evidence writer so it only records `run_card` when the RunCard file actually exists. This keeps Wave 4 backfill from emitting dangling disclosure refs for runs without a decision-backed card while preserving the seeded canonical runs that do emit RunCards.
